e2e/testcases/invalid_git_branch_test.go

```diff
@@ -58,6 +60,30 @@ func TestInvalidRootSyncBranchStatus(t *testing.T) {
 
 func TestInvalidRepoSyncBranchStatus(t *testing.T) {
 	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipMonoRepo, ntopts.NamespaceRepo(namespaceRepo, configsync.RepoSyncName))
+	nt.T.Cleanup(func() {
```

Comment:
I don't see a need to reset the RepoSync branch.

If the test failed:
1. before setting the invalid branch, no need to cleanup as the RepoSync is not created yet.
2. after setting the invalid branch, but before setting back to main. No resources will be managed by RepoSync and the RootSync will be reset by the test infra, which prunes the invalid RepoSync at the end.
3. after setting back to main, no need to reset again.

Reply:
> No resources will be managed by RepoSync and the RootSync will be reset by the test infra

This is not correct. The general reset does not change the local checkout branch. So if you last checked out some other branch for this repo, it won't reset correctly unless you check out the right branch. The generic reset also does not reset the branch in the the RootSync or RepoSync.

Reply:
The general reset will reset all repos before and after each test case. Every test case will start with a new initialized git repository. In the [resetRepository](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/nomostest/config-sync.go#L1248) function, both `ReInit` and `NewRepository` do a `git init` followed by `git checkout -b main`, which guarantees that each repo starts with the main branch.

Reply:
Removed the cleanup code for this PR.